// @shopify/storefront-kit-react/dist/node-dev/BuyNowButton.js
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const require$$0 = require("react");
const CartProvider = require("./CartProvider.js");
const BaseButton = require("./BaseButton.js");
function BuyNowButton(props) {
  const { cartCreate, checkoutUrl } = CartProvider.useCart();
  const [loading, setLoading] = require$$0.useState(false);
  const {
    quantity,
    variantId,
    onClick,
    attributes,
    children,
    ...passthroughProps
  } = props;
  require$$0.useEffect(() => {
    if (checkoutUrl) {
      window.location.href = checkoutUrl;
    }
  }, [checkoutUrl]);
  const handleBuyNow = require$$0.useCallback(() => {
    setLoading(true);
    cartCreate({
      lines: [
        {
          quantity: quantity ?? 1,
          merchandiseId: variantId,
          attributes
        }
      ]
    });
  }, [cartCreate, quantity, variantId, attributes]);
  return /* @__PURE__ */ jsxRuntime.jsx(
    BaseButton.BaseButton,
    {
      disabled: loading ?? passthroughProps.disabled,
      ...passthroughProps,
      onClick,
      defaultOnClick: handleBuyNow,
      children
    }
  );
}
exports.BuyNowButton = BuyNowButton;
